src/features/surveys/components/SurveyEditor/blocks/TextBlock.tsx

```diff
@@ -1,13 +1,117 @@
-import { Box } from '@mui/material';
-import { FC } from 'react';
+import { Box, ClickAwayListener, TextField, Typography } from '@mui/material';
+import { FC, KeyboardEvent, useEffect, useRef, useState } from 'react';
+import { FormattedMessage as Msg, useIntl } from 'react-intl';
+
+import theme from 'theme';
 import { ZetkinSurveyTextElement } from 'utils/types/zetkin';
 
 interface TextBlockProps {
+  inEditMode: boolean;
   element: ZetkinSurveyTextElement;
+  onEditModeEnter: () => void;
+  onEditModeExit: (textBlock: ZetkinSurveyTextElement['text_block']) => void;
 }
 
-const TextBlock: FC<TextBlockProps> = ({ element }) => {
-  return <Box>{JSON.stringify(element)}</Box>;
+const TextBlock: FC<TextBlockProps> = ({
+  inEditMode,
+  element,
+  onEditModeEnter,
+  onEditModeExit,
+}) => {
+  const intl = useIntl();
+  const [header, setHeader] = useState(element.text_block.header);
+  const [content, setContent] = useState(element.text_block.content);
+
+  const [focusHeader, setFocusHeader] = useState(true);
+  const [focusContent, setFocusContent] = useState(false);
```

Comment:
These make me hesitate a little bit, because they are essentially derived state. You can only focus one of these at a time, but it's technically possible to set both to `true`, meaning that there is a bug brewing here.

Would it be a viable option to replace these two states with a single `useState<'header' | 'content' | null>(null);` and then set it to either `'header'` or `'content'` when an element is clicked??

Reply:
Great suggestion, fixing it like this!